src/activities/Associations.js

```diff
@@ -0,0 +1,25 @@
+import { Entity } from '../es6/Entity.js';
+import { Classes } from '../hypermedia-constants';
+
+/**
+ * Associations class representation of  associations.
+ */
+
+export class Associations extends Entity {
+
+	getSingleAssocationHrefs() {
+		if (!this._entity) {
+			return [];
+		}
+
+		const singleAssociations = this._entity.getSubEntitiesByClass(Classes.associations.singleAssociation);
+		const singleAssociationsHrefs = [];
+		singleAssociations.forEach(a => {

```

Comment:
Consider: `singleAssociations.map( a => a.getLinkByRel('self').href )` if you are going to do the unconditional `href` property access here (seems pretty safe for a `self` rel).